### PR TITLE
Add alternative solution to TwelveDays exercise

### PR DESCRIPTION
**Pull request by samWson**
Here is how an alternative solution to Twelve Days might look based on the solution shown in #386. The biggest change is that `#verse:` is swapped out for `#getVerse:` (might keep the name the same if we use this solution). `#verses` and `#dayName` are direct copies of the first solutions `#gifts` and `#ordinal:` respectively, the names just changed to fit in with the alternate solution. 

The big point of this solution is that it uses `SequencableCollection#>>joinUsing:last:` in `#getVerse`.

Let me know if you want to go ahead with this.

### CODE REVIEW

File: dev/src/Exercise@TwelveDays/TwelveDaysTest.class.st [hunk 1]
```diff
@@ -86,7 +86,7 @@ TwelveDaysTest class >> version [
 { #category : #running }
 TwelveDaysTest >> setUp [
```
Comment (macta) [inline dev/src/Exercise@TwelveDays/TwelveDaysTest.class.st]: Shouldn't this method be in TwelveDays2Test - e.g. you are overriding the setup from the parent tests, but inheriting the actual tests? This way, the original class is no longer tested?

Comment (samWson) [reply dev/src/Exercise@TwelveDays/TwelveDaysTest.class.st]: Odd but somehow `TwelveDays2Test>>#setup` was not a part of this commit.

**Comment by samWson** [comment]
@macta will I need to generate the meta data for this additional solution i.e. as in [publishing a completed exercise](https://github.com/exercism/pharo-smalltalk#publishing-a-completed-exercise-maintainers)?

**Comment by macta** [comment]
No - the exercises are only visible to mentors and if we choose to publish as if we were students ourselves.

The publishing step only effects making the tests available and we derive the readme from the test class as well.

File: dev/src/Exercise@TwelveDays/TwelveDaysTest.class.st (hunk 1 above)
Comment (samWson) [reply dev/src/Exercise@TwelveDays/TwelveDaysTest.class.st]: @macta are you happy with the changes? Let me know and we can get this merged.

**Comment by samWson** [comment]
@bencoman I'd like to get a few of these open pull requests merged. PR's like this should not require regeneration of the tests and meta data. Will it cause problems if I merge this one (and others) before the work on #372 is merged?

**Comment by bencoman** [comment]
@samWson This PR won't require regeneration.  Without refreshing myself with the generation code, I'm not sure if TwelveDays2Test would survive regeneration, but we should fix that if it does.

**Pull request merged**